watertap/unit_models/ion_exchange_0D.py

```diff
@@ -0,0 +1,1352 @@
+    ControlVolume0DBlock,
+    declare_process_block_class,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    MomentumBalanceType,
+    UnitModelBlockData,
+    useDefault,
+    MaterialFlowBasis,
+)
+from idaes.core.util import get_solver
+from idaes.core.util.tables import create_stream_table_dataframe
+from idaes.core.util.constants import Constants
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.exceptions import ConfigurationError
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+from idaes.core.util.math import smooth_max
+
+
+_log = idaeslog.getLogger(__name__)
+
+__author__ = "Kurban Sitterley"
+
+
+@declare_process_block_class("IonExchange0D")
+class IonExchangeODData(UnitModelBlockData):
+    """
+    Zero order ion exchange model
+    """
+
+    # CONFIG are options for the unit model, this simple model only has the mandatory config options
+    CONFIG = ConfigBlock()
+
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([False]),
+            default=False,
+            description="Dynamic model flag - must be False",
+            doc="""Indicates whether this model will be dynamic or not,
+    **default** = False. The filtration unit does not support dynamic
+    behavior, thus this must be False.""",
+        ),
+    )
+
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Indicates whether holdup terms should be constructed or not.
+    **default** - False. The filtration unit does not have defined volume, thus
+    this must be False.""",
+        ),
+    )
+
+    CONFIG.declare(
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    CONFIG.declare(
+        "target_ion",
+        ConfigValue(default="Ca_2+", domain=str, description="Target Ion"),
+    )
+
+    CONFIG.declare(
+        "regenerant",
+        ConfigValue(default="HCl", domain=str, description="Regenerant"),
+    )
+
```

Comment:
I geared this model so that IX really could be any chemical. Presumably, the target ion (or component) would be part of the larger ``solute_list`` - so is it possible to make a ``Var`` a ``str``? Or how else could I have the user designate a target ion/compound?

Reply:
The target ions can be designated through the property package. This comment was on whether the regenerate agent should be limited to certain specific chemicals (e.g., HCl or another acid).

Reply:
How can they be desginated through the property package if not through CONFIG though?

Reply:
With a second read,  is your target ion CONFIG to determine which ion to be *selectively* exchanged?   OK to do it this way if this is your point.  but I would wonder, if the solution contains multiple ions, whether you can determine what the target ion is just by choosing a config.  My idea is that the selectivity can be modeled by affinity or material selectivity data for different ions.   In that case, only property block data is needed.  Another suggestion was to make the config domain more specific, something like `In( )` to assign a candidate list.  `Str` would be too loose.  Same idea for the regenerate agent CONFIG.    Depending on your goal for this version, these comments can be treated at your discretion.